dockerfiles/base/scripts/base/startup.sh

```diff
@@ -5,417 +5,7 @@
-  REFERENCE_CONTAINER_ENVIRONMENT_FILE="${CHE_CONTAINER_CONFIG}/${CHE_ENVIRONMENT_FILE}"
-  REFERENCE_CONTAINER_COMPOSE_FILE="${CHE_CONTAINER_INSTANCE}/${CHE_COMPOSE_FILE}"
-  REFERENCE_CONTAINER_COMPOSE_HOST_FILE="${CHE_CONTAINER_INSTANCE}/${CHE_HOST_COMPOSE_FILE}"
-
-  CHE_CONTAINER_OFFLINE_FOLDER="${CHE_CONTAINER_BACKUP}"
-  CHE_HOST_OFFLINE_FOLDER="${CHE_HOST_BACKUP}"
-
-  CHE_HOST_CONFIG_MANIFESTS_FOLDER="${CHE_HOST_INSTANCE}/manifests"
-  CHE_CONTAINER_CONFIG_MANIFESTS_FOLDER="${CHE_CONTAINER_INSTANCE}/manifests"
-
-  CHE_HOST_CONFIG_MODULES_FOLDER="${CHE_HOST_INSTANCE}/modules"
-  CHE_CONTAINER_CONFIG_MODULES_FOLDER="${CHE_CONTAINER_INSTANCE}/modules"
-
-  # TODO: Change this to use the current folder or perhaps ~?
-  if is_boot2docker && has_docker_for_windows_client; then
-    if [[ "${CHE_HOST_INSTANCE,,}" != *"${USERPROFILE,,}"* ]]; then
-      CHE_HOST_INSTANCE=$(get_mount_path "${USERPROFILE}/.${CHE_MINI_PRODUCT_NAME}/")
-      warning "Boot2docker for Windows - CHE_INSTANCE set to $CHE_HOST_INSTANCE"
-    fi
-    if [[ "${CHE_HOST_CONFIG,,}" != *"${USERPROFILE,,}"* ]]; then
-      CHE_HOST_CONFIG=$(get_mount_path "${USERPROFILE}/.${CHE_MINI_PRODUCT_NAME}/")
-      warning "Boot2docker for Windows - CHE_CONFIG set to $CHE_HOST_CONFIG"
-    fi
-  fi
-
-  # Special function to perform special behaviors if you are running nightly version
-  verify_nightly_accuracy
-
-  # Do not perform a version compatibility check if running upgrade command.
-  # The upgrade command has its own internal checks for version compatibility.
-  if [ $1 != "upgrade" ]; then
-  	if [[ "${FAST_BOOT}" = "false" ]]; then
-      verify_version_compatibility
-    else
-      warning "Skipping version compatibility check..."
-    fi
-  else
-    verify_version_upgrade_compatibility
-  fi
-}
-
-cleanup() {
-  RETURN_CODE=$?
-
-  # CLI developers should only return '3' in code after the init() method has completed.
-  # This will check to see if the CLI directory is not mounted and only offer the error
-  # message if it isn't currently mounted.
-  if [ $RETURN_CODE -eq "3" ]; then
-    error ""
-    error "Unexpected exit: Trace output saved to $CHE_HOST_CONFIG/cli.log."
-  fi
-}
-
-start() {
-  # Bootstrap networking, docker, logging, and ability to load cli.sh and cli-functions.sh
-  init "$@"
-
-  # Removes "--fast" and "--debug" from the positional arguments if it is set.
-  set -- "${@/\-\-fast/}"
-  set -- "${@/\-\-debug/}"
-  
-  # Begin product-specific CLI calls
-  info "cli" "Loading cli..."
-
-
-  # The pre_init method is unique to each assembly. This method must be provided by 
-  # a custom CLI assembly in their container and can set global variables which are 
-  # specific to that implementation of the CLI. This method must be called after
-  # networking has been established and initial images downloaded.
-  cli_pre_init
-
-  cli_init "$@"
-  cli_parse "$@"
-  cli_execute "$@"
-}
+source /scripts/base/startup_funcs.sh
```

Comment:
@TylerJewell those file names are changed as you requested in comment above, I move cli/cli-functions.sh to library.sh and we are using short notation "funcs" instead of "functions" and underscore instead of dash.

Regarding mounting scripts, it is still preserved. The tests uses different image and mount a local location of a scripts to the same location inside the test container and the test container then mount this location to /base/scripts, so the tests run with local scripts, which are mounted to the container. You can see it e.g. at https://github.com/eclipse/che/pull/3700/files#diff-1786fd5485e1f71d9b8ab5197898dcdbR42 where SCRIPTS_DIR is mounted to /scripts/base. This approach is used to test code in its current state.